aws/resource_aws_datapipeline_test.go

```diff
@@ -0,0 +1,269 @@
+	rName := fmt.Sprintf("tf-datapipeline-%s", acctest.RandString(5))
+	resourceName := "aws_datapipeline.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSDataPipelineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDataPipelineConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					testAccCheckAWSDataPipelineDisappears(&conf),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSDataPipeline_tags(t *testing.T) {
+	var conf datapipeline.PipelineDescription
+	rName := acctest.RandString(5)
+	resourceName := "aws_datapipeline.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSDataPipelineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDataPipelineConfigWithTags(rName, "foo", "bar", "fizz", "buzz"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.foo", "bar"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.fizz", "buzz"),
+				),
+			},
+			{
+				Config: testAccAWSDataPipelineConfigWithTags(rName, "foo", "bar2", "fizz2", "buzz2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.foo", "bar2"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.fizz2", "buzz2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSDataPipelineConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSDataPipelineDisappears(conf *datapipeline.PipelineDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+		params := &datapipeline.DeletePipelineInput{
+			PipelineId: conf.PipelineId,
+		}
+
+		_, err := conn.DeletePipeline(params)
+		if err != nil {
+			return err
+		}
+		return waitForDataPipelineDeletion(conn, *conf.PipelineId)
+	}
+}
+
+func testAccCheckAWSDataPipelineDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_datapipeline" {
+			continue
+		}
+		// Try to find the Pipeline
```

Comment:
It seems like this logic should be replaced with `resourceAwsDataPipelineRetrieve()` 👍

Reply:
fixed